### PR TITLE
Update documentation about nested directories components

### PR DESCRIPTION
**Pull request by riochandra1212**
Migration nuxt/component v1 to v2 makes a great feature about nested directories, starting `nuxt@2.15` using `@nuxt/component v2`. [nested directories nuxt component](https://github.com/nuxt/components#nested-components)

at first, I got problem at nuxt production that update nuxt/component to v2. Many component that not called directly.

I think nuxt-newbie will get this problem when learn it for the first time, so i decide to edit documentation about this update.

I edit documentation about nuxt [nested directories component](https://nuxtjs.org/docs/2.x/directory-structure/components#nested-directories) relate to new update @nuxt/component version 2.

### CODE REVIEW

File: content/en/guides/directory-structure/components.md [hunk 1]
```diff
@@ -153,46 +153,40 @@ If you have components in nested directories such as:
...
 ```bash{}[nuxt.config.js]
 components: {
   dirs: [
-    '~/components',
+    '~/components', // shortcut to { path: '~/components' }
       {
-        path: '~/components/base/',
-        prefix: 'Base'
+        path: '~/components/base/foo/',
+        prefix: 'awesome'
```
Comment (pi0) [inline content/en/guides/directory-structure/components.md:181]: Default prefix is (with directory name) is to ensure fs path matches component name. This way we can quickly find component based on it's name. `prefix` is an advanced option and mostly usable when creating component libraries. For this docs section, i would suggest simplify example by registering `~/components/base` to remove `Base` prefix.

Comment (riochandra1212) [reply content/en/guides/directory-structure/components.md:181]: Ok, that's make sense, i'll fix that

**Comment by riochandra1212** [comment]
@pi0 I've fix default prefix that you mention earlier, may you review it.

**Comment by pi0** [comment]
Thanks @riochandra1212. I've added some changes to explain simplest way to remove prefix. Advanced `prefix` and `pathPrefix` options are explained in longer docs linked in the bottom. We will move all docs into website soon (see #1244)

**Review by pi0 — APPROVED** [review]

**Pull request merged**